packages/gen-ai-hub/src/client/openai/openai-client.ts

```diff
@@ -18,20 +17,22 @@ const apiVersion = '2024-02-01';
  * OpenAI Client.
  */
 export class OpenAiClient {
+  // TODO: document constructor
+  //* @param modelDeployment - This configuration is used to retrieve a deployment. Depending on the configuration use either the given deployment ID or the model name to retrieve matching deployments. If model and deployment ID are given, the model is verified against the deployment.
+  constructor(private modelDeployment: ModelDeployment<OpenAiChatModel>) {}
```

Comment:
I left the deployment ID config in here as it was already part of the model and required less changes. If we decide to not allow setting it explicitly I will make a proposal without.